bundle/Command/TagContentByTypesCommand.php

```diff
@@ -0,0 +1,198 @@
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output): void
+    {
+        $this->input = $input;
+        $this->style = new SymfonyStyle($input, $output);
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $query = new Query();
+        $query->filter = new Criterion\LogicalAnd([
+            new Criterion\ParentLocationId($this->getParentLocation()),
+            new Criterion\ContentTypeIdentifier($this->getContentTypes()),
+        ]);
+
+        $batchSize = 50;
+
+        $searchResults = $this->repository->getSearchService()->findContent($query);
+        $totalResults = $searchResults->totalCount;
+
+        $this->style->newLine();
+        $this->style->progressStart($totalResults);
+
+        $this->repository->beginTransaction();
+
+        for ($offset = 0; $offset < $totalResults; $offset += $batchSize) {
+            $query->offset = $offset;
+            $query->limit = $batchSize;
+            $searchResults = $this->repository->getSearchService()->findContent($query);
+
+            if ($this->input->getOption('field-identifiers') === null) {
+                $fieldIdentifiers = $this->configResolver->getParameter('tag_command_default_field_identifiers', 'ngsite');
+            } else {
+                $fieldIdentifiers = $this->getFieldIdentifiers();
+            }
+
+            $this->repository->beginTransaction();
+
+            foreach ($searchResults->searchHits as $searchHit) {
+                $result = $this->repository->sudo(
+                    function () use ($searchHit, $fieldIdentifiers): ?int {
+                        /** @var \Ibexa\Contracts\Core\Repository\Values\Content\Content $content */
+                        $content = $searchHit->valueObject;
+
+                        $contentDraft = $this->repository->getContentService()->createContentDraft($content->contentInfo);
+                        $contentUpdateStruct = $this->repository->getContentService()->newContentUpdateStruct();
+
+                        foreach ($fieldIdentifiers as $fieldIdentifier) {
+                            if (!$this->hasField($content, $fieldIdentifier)) {
+                                continue;
+                            }
+
+                            if (!$content->getField($fieldIdentifier)->value instanceof TagFieldValue) {
+                                $this->style->error(sprintf('Field with identifier %s must be a type of eztags', $fieldIdentifier));
+                                return Command::FAILURE;
+                            }
+
+                            $alreadyAssignedTags = $content->getFieldValue($fieldIdentifier)->tags;
+                            $tagsToAssign = array_filter($alreadyAssignedTags, fn ($alreadyAssignedTag) => $this->getTag()->id !== $alreadyAssignedTag->id);
+                            $tagsToAssign[] = $this->getTag();
```

Comment:
We are using `$this->getTag()` in two consecutive lines of code, can we not call this just once?

Reply:
solved in efdb0ccb